domain/application/state/storage_test.go

```diff
@@ -94,6 +94,77 @@ INSERT INTO storage_pool (uuid, name, type) VALUES (?, ?, ?)
 	return poolUUID
 }
 
+func (s *applicationStateSuite) TestGetApplicationStorageDirectivesInfo(c *tc.C) {
```

Comment:
On the testing front it would be nice to have a test asserting the not found condition. The reasons for this are the same as the ones listed above. We must ensure we maintain contracts and expectations over a long period of time and give ourselves confidence in future changes.

This is tedious but it contributes to a stable and predictable Juju for the long term.